src/common/SmartUrlInput.js

```diff
@@ -113,26 +110,11 @@ export default function SmartUrlInput(props: Props): Node {
   const handleChange = useCallback(
     (_value: string) => {
       setValue(_value);
-
-      onChangeText(
-        fixRealmUrl(autocompleteRealm(_value, { protocol: 'https://', domain: 'zulipchat.com' })),
```

Comment:
These little helper functions exist only for the sake of this component, and are closely tied to the details of its UI. So when deleting these references to them, let's be sure to celebrate by deleting the functions themselves.